lib/prawn/embedded_files/embedded_file.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+require 'digest/md5'
+
+module Prawn
+  module EmbeddedFiles
+    class EmbeddedFile
+      attr_reader :chksum
+
+      def initialize(data, options = {})
+        @creation_date = options[:creation_date]
+        unless @creation_date.instance_of?(Time)
+          @creation_date = Time.now.utc
+        end
+
+        @mod_date = options[:modification_date]
+        unless @mod_date.instance_of?(Time)
+          @mod_date = Time.now.utc
+        end
+
+        @chksum = Digest::MD5.hexdigest(data)
```

Comment:
You are using `#hexdigest` here where you should be using `#digest`. Otherwise the value isn't the expected 16 byte digest string.

Reply:
Ok, thanks for the clarification